### PR TITLE
Add rename function

### PR DESCRIPTION
**Pull request by OFR-IIASA**

### CODE REVIEW

File: pyam/core.py [hunk 1]
```diff
@@ -405,6 +405,28 @@ def validate(self, criteria={}, exclude=False):
...
+        ----------
+        mapping: dict
+            for each column where entries should be renamed, provide current name and target name
+            {<column name>: {<current_name_1>: <target_name_1>, <current_name_2>: <target_name_2>}}
+        inplace: bool, default False
+            if True, do operation inplace and return None
+        """
+        ret = copy.deepcopy(self) if not inplace else self
+        for col in mapping:
+            if col in ['region', 'variable', 'unit']:
```
Comment (gidden) [inline pyam/core.py]: This can be `if not and save an indention

File: pyam/core.py [hunk 2]
```diff
@@ -405,6 +405,28 @@ def validate(self, criteria={}, exclude=False):
...
+                ret.data.loc[:, col] = self.data.loc[:, col].replace(mapping[col])
+            else:
+                raise ValueError('renaming by {} not supported!'.format(col))
+
+        ret.data = ret.data.groupby(IAMC_IDX + ['year']).sum().reset_index()
```
Comment (gidden) [inline pyam/core.py]: @danielhuppman this is your YEAR_IDX right?

Comment (danielhuppmann) [reply pyam/core.py]: not quite, `YEAR_IDX` as proposed in #52 (not yet merged) is again different - we should probably come up with a better structure for the index-column collections at some point down the line...

Comment (danielhuppmann) [reply pyam/core.py]: How about ``LONG_IDX = IAMC_IDX + ['year']`` (i.e., `['model', 'scenario', 'region', 'variable', 'unit', 'year']` for all columns except `value`) in the `data` table?

**Comment by gidden** [comment]
thanks @OFR-IIASA !

**Pull request merged**